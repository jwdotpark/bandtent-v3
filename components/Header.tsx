import React from 'react'
import Link from 'next/link'
import { Router, useRouter } from 'next/router'
import { signOut, useSession } from 'next-auth/react'
import ColorButton from '../components/misc/ColorButton'
import {
  Stack,
  Flex,
  Spacer,
  Box,
  Text,
  Button,
  Image,
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
} from '@chakra-ui/react'
import { Media } from '../utils/media'
import { ChevronDownIcon } from '@chakra-ui/icons'

const Header: React.FC = () => {
  const router = useRouter()
  const isActive: (pathname: string) => boolean = (pathname) =>
    router.pathname === pathname

  const { data: session, status } = useSession()

  let left = (
    <Box>
      <Link href="/">
        <Box className="bold" data-active={isActive('/')}>
          <Button fontSize="2xl">Feed</Button>
        </Box>
      </Link>
    </Box>
  )

  let right = null

  if (status === 'loading') {
    right = (
      <Box mr="auto">
        <Text>Validating session ...</Text>
      </Box>
    )
  }

  if (!session) {
    right = (
      <Box>
        <Button left="0" data-active={isActive('/signup')}>
          <Link href="/api/auth/signin">Log in </Link>
        </Button>
      </Box>
    )
  }

  if (session) {
    left = (
      <div className="left">
        <Link href="/">
          <a className="bold" data-active={isActive('/')}>
            Feed
          </a>
        </Link>
        <Link href="/drafts">
          <a data-active={isActive('/drafts')}>My drafts</a>
        </Link>
      </div>
    )
    right = (
      <div className="right">
        <p>
          {session.user.name} ({session.user.email})
        </p>
        <Link href="/create">
          <button>
            <a>New post</a>
          </button>
        </Link>
        <button
          onClick={() => {
            signOut()
            router.push('/')
          }}
        >
          <a>Log out</a>
        </button>
      </div>
    )
  }

  return (
    <nav>
      {/* desktop */}
      <Media greaterThanOrEqual="md">
        <Flex borderBottom="1px solid gray" mx="2">
          {/* left */}
          <Box p="2">
            <Link href="/">
              <Box className="bold" data-active={isActive('/')}>
                <Button size="sm">Feed</Button>
              </Box>
            </Link>
          </Box>
          <Spacer />
          {/* right */}
          {/* no login */}
          {!session && (
            <>
              <Stack direction="row" p="2">
                <Link href="/api/auth/signin">
                  <Button size="sm">
                    {status === 'loading' ? 'Validating Session..' : 'Log In'}
                  </Button>
                </Link>
                <ColorButton />
              </Stack>
            </>
          )}
          {/* yes login */}
          {session && (
            <>
              <Stack direction="row" p="2">
                <Link href="/auth/me">
                  <Button size="sm">
                    {session.user.name ? session.user.name : session.user.email}
                  </Button>
                </Link>

                {/* <Button size="sm" data-active={isActive('/drafts')}> */}
                <Button size="sm">
                  <Link href="/drafts">My Drafts</Link>
                </Button>

                <Link href="/create">
                  <Button size="sm">
                    <Text>Add</Text>
                  </Button>
                </Link>
                <Button
                  size="sm"
                  onClick={() => {
                    signOut()
                    router.push('/')
                  }}
                >
                  <Text>Log out</Text>
                </Button>
                <ColorButton />
              </Stack>
            </>
          )}
        </Flex>
      </Media>

      {/* mobile */}
      <Media lessThan="md">
        <Flex borderBottom="1px solid gray" mx="2">
          {/* left */}
          <Box p="2">
            <Link href="/">
              <Box className="bold" data-active={isActive('/')}>
                <Button size="sm">Feed</Button>
              </Box>
            </Link>
          </Box>
          <Spacer />
          {/* right */}

          {/* no login */}
          {!session && (
            <>
              <Stack direction="row" p="2">
                <Link href="/api/auth/signin">
                  <Button size="sm">
                    {status === 'loading' ? 'Validating Session..' : 'Log In'}
                  </Button>
                </Link>
                <ColorButton />
              </Stack>
            </>
          )}
          {/* yes login */}
          {session && (
            <>
              <Box m="2" mx="auto">
                <Menu>
                  <MenuButton
                    as={Button}
                    size="sm"
                    rightIcon={<ChevronDownIcon />}
                  >
                    <Text>{session.user.name}</Text>
                  </MenuButton>
                  <MenuList>
                    <Link href="/auth/me">
                      <MenuItem>Me</MenuItem>
                    </Link>
                    <Link href="/drafts">
                      <MenuItem>My Drafts</MenuItem>
                    </Link>
                    <Link href="/create">
                      <MenuItem>
                        <Text>Add</Text>
                      </MenuItem>
                    </Link>
                  </MenuList>
                </Menu>
              </Box>

              <Stack direction="row" p="2">
                <Button
                  size="sm"
                  onClick={() => {
                    signOut()
                    router.push('/')
                  }}
                >
                  <Text>Log out</Text>
                </Button>
                <ColorButton />
              </Stack>
            </>
          )}
        </Flex>
      </Media>
    </nav>
  )
}

export default Header
