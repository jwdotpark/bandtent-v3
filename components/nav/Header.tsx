import React from 'react'
import Link from 'next/link'
import { useRouter } from 'next/router'
import { signOut, useSession } from 'next-auth/react'
import ColorButton from '../misc/ColorButton'
import {
  Stack,
  Flex,
  Spacer,
  Box,
  Text,
  Button,
  Image,
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
} from '@chakra-ui/react'
import { Media } from '../../utils/media'
import SearchButton from './SearchButton'
import {
  ChevronDownIcon,
  AddIcon,
  PlusSquareIcon,
  TimeIcon,
  LockIcon,
  UnlockIcon,
  HamburgerIcon,
} from '@chakra-ui/icons'

const Header: React.FC = () => {
  const router = useRouter()
  const isActive: (pathname: string) => boolean = (pathname) =>
    router.pathname === pathname

  const { data: session, status } = useSession()

  return (
    <nav>
      {/* desktop */}
      <Media greaterThanOrEqual="md">
        <Flex borderBottom="1px solid gray" mx="2">
          {/* left */}
          <Stack direction="row" p="2">
            <Link href="/">
              <Box className="bold" data-active={isActive('/')}>
                <Button leftIcon={<HamburgerIcon />} size="sm" boxShadow="md">
                  Feed
                </Button>
              </Box>
            </Link>
            {/* search */}
            <SearchButton />
            {session && (
              <Link href="/auth/me">
                <Button
                  boxShadow="md"
                  size="sm"
                  leftIcon={
                    <Image
                      boxSize="25px"
                      display="inline"
                      border="1px inset  gray"
                      src={session.user.image}
                      fallbackSrc="https://picsum.photos/200"
                      borderRadius="full"
                      alt={session.user.name}
                    />
                  }
                >
                  {session.user.name ? session.user.name : session.user.email}
                </Button>
              </Link>
            )}
          </Stack>
          <Spacer />
          {/* right */}
          {/* no login */}
          {!session && (
            <>
              <Stack direction="row" p="2">
                <Link href="/api/auth/signin">
                  <Button size="sm" leftIcon={<UnlockIcon />}>
                    {status === 'loading' ? 'Validating Session..' : 'Log In'}
                  </Button>
                </Link>
                <ColorButton />
              </Stack>
            </>
          )}
          {/* yes login */}
          {session && (
            <>
              <Stack direction="row" p="2">
                <Link href="/create">
                  <Button
                    boxShadow="md"
                    size="sm"
                    leftIcon={<PlusSquareIcon />}
                  >
                    <Text>Add</Text>
                  </Button>
                </Link>
                <Button boxShadow="md" size="sm" leftIcon={<TimeIcon />}>
                  <Link href="/drafts">My Drafts</Link>
                </Button>
                <Button
                  boxShadow="md"
                  leftIcon={<LockIcon />}
                  size="sm"
                  onClick={() => {
                    signOut()
                    router.push('/')
                  }}
                >
                  <Text>Log out</Text>
                </Button>
                <Box boxShadow="md">
                  <ColorButton />
                </Box>
              </Stack>
            </>
          )}
        </Flex>
      </Media>

      {/* mobile */}
      <Media lessThan="md">
        <Flex borderBottom="1px solid gray" mx="2">
          {/* left */}
          <Box p="2">
            <Link href="/">
              <Box className="bold" data-active={isActive('/')}>
                <Button leftIcon={<HamburgerIcon />} size="sm">
                  Feed
                </Button>
              </Box>
            </Link>
          </Box>
          <Spacer />
          {/* right */}

          {/* no login */}
          {!session && (
            <>
              <Stack direction="row" p="2">
                <Link href="/api/auth/signin">
                  <Button size="sm">
                    {status === 'loading' ? 'Validating Session..' : 'Log In'}
                  </Button>
                </Link>
                <ColorButton />
              </Stack>
            </>
          )}
          {/* yes login */}
          {session && (
            <>
              <Box m="2" mx="auto">
                <Menu matchWidth={true}>
                  <MenuButton
                    as={Button}
                    size="sm"
                    rightIcon={<ChevronDownIcon />}
                  >
                    <Text>{session.user.name}</Text>
                  </MenuButton>
                  <MenuList>
                    <Link href="/auth/me">
                      <MenuItem
                        icon={
                          <Image
                            sx={{ transform: 'translateX(-3px)' }}
                            border="1px solid gray"
                            src={session.user.image}
                            boxSize="20px"
                            borderRadius="md"
                            alt={session.user.name}
                          />
                        }
                      >
                        <Text>Profile</Text>
                      </MenuItem>
                    </Link>
                    <Link href="/create">
                      <MenuItem icon={<AddIcon mr="2" />}>
                        <Text>Add</Text>
                      </MenuItem>
                    </Link>
                    <Link href="/drafts">
                      <MenuItem icon={<PlusSquareIcon mr="2" />}>
                        My Drafts
                      </MenuItem>
                    </Link>
                    <MenuItem
                      onClick={() => {
                        signOut()
                        router.push('/')
                      }}
                      icon={<LockIcon mr="2" />}
                    >
                      <Text>Log out</Text>
                    </MenuItem>
                  </MenuList>
                </Menu>
              </Box>

              <Stack direction="row" p="2">
                <ColorButton />
              </Stack>
            </>
          )}
        </Flex>
      </Media>
    </nav>
  )
}

export default Header
