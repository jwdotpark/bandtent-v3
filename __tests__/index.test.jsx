import { render, screen } from './test-utils'
import Header from '../components/Layout'

describe('NavBar Component', () => {
  it('can see add button when has session', async () => {
    render(<Header />)

    const feed = await screen.findByText('Feed')
    expect(feed).toBeInTheDocument()

    
  })
})
