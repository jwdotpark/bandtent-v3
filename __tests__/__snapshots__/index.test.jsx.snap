// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component Show Log Out when has session 1`] = `
<div>
  <nav>
    <header
      class="css-xmxh5i"
    >
      <div
        class="chakra-stack css-83r2v3"
      >
        <div>
          <div
            class="bold css-0"
            href="/"
          >
            <button
              class="chakra-button css-1a4ohpf"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-12x3nr5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <p
                class="chakra-text css-1kyf6sp"
              >
                Feed
              </p>
            </button>
          </div>
        </div>
        <div>
          <button
            class="chakra-button css-1a4ohpf"
            href="/auth/undefined"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <img
                aria-hidden="true"
                class="chakra-image__placeholder css-6fak1j"
                focusable="false"
                src="https://picsum.photos/200"
              />
            </span>
            <p
              class="chakra-text css-1lv3xn0"
            />
          </button>
        </div>
        <div>
          <div
            class="bold css-0"
          >
            <form>
              <div
                class="chakra-input__group css-4302v8"
              >
                <div
                  class="chakra-input__left-element css-1u2k0bl"
                >
                  <svg
                    class="chakra-icon css-onkibi"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <input
                  class="chakra-input css-yelcpu"
                  placeholder="Search"
                  type="text"
                />
                <div
                  class="chakra-input__right-element css-1uc557g"
                >
                  <button
                    class="chakra-button css-g7mvkz"
                    type="submit"
                  >
                    <svg
                      class="chakra-icon css-onkibi"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        fill="currentColor"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="chakra-stack css-83r2v3"
      >
        <div>
          <button
            class="chakra-button css-1a4ohpf"
            href="/create"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-j4318f"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="2"
                >
                  <rect
                    height="18"
                    rx="2"
                    ry="2"
                    width="18"
                    x="3"
                    y="3"
                  />
                  <path
                    d="M12 8v8"
                  />
                  <path
                    d="M8 12h8"
                  />
                </g>
              </svg>
            </span>
            <p
              class="chakra-text css-6iqdsv"
            >
              Add
            </p>
          </button>
        </div>
        <div>
          <button
            class="chakra-button css-1a4ohpf"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-nygapn"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  fill="currentColor"
                >
                  <path
                    d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm0,22A10,10,0,1,1,22,12,10.011,10.011,0,0,1,12,22Z"
                  />
                  <path
                    d="M17.134,15.81,12.5,11.561V6.5a1,1,0,0,0-2,0V12a1,1,0,0,0,.324.738l4.959,4.545a1.01,1.01,0,0,0,1.413-.061A1,1,0,0,0,17.134,15.81Z"
                  />
                </g>
              </svg>
            </span>
            <p
              class="chakra-text css-1749hm6"
              href="/drafts"
            />
          </button>
        </div>
        <div>
          <button
            class="chakra-button css-1a4ohpf"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-1p86u1z"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.5,9.5h-.75V6.75a6.75,6.75,0,0,0-13.5,0V9.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5ZM7.75,6.75a4.25,4.25,0,0,1,8.5,0V9a.5.5,0,0,1-.5.5H8.25a.5.5,0,0,1-.5-.5Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <p
              class="chakra-text css-1ip116p"
            >
              Log out
            </p>
          </button>
        </div>
        <div>
          <div
            class="css-17hs7hi"
          >
            <button
              class="chakra-button css-5c2176"
              type="button"
            >
              <svg
                class="chakra-icon css-nyh0wu"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="5"
                  />
                  <path
                    d="M12 1v2"
                  />
                  <path
                    d="M12 21v2"
                  />
                  <path
                    d="M4.22 4.22l1.42 1.42"
                  />
                  <path
                    d="M18.36 18.36l1.42 1.42"
                  />
                  <path
                    d="M1 12h2"
                  />
                  <path
                    d="M21 12h2"
                  />
                  <path
                    d="M4.22 19.78l1.42-1.42"
                  />
                  <path
                    d="M18.36 5.64l1.42-1.42"
                  />
                </g>
              </svg>
            </button>
          </div>
        </div>
      </div>
    </header>
  </nav>
  <div
    class="css-11rrnlf"
  />
</div>
`;

exports[`NavBar Component can see add button when has session 1`] = `
<div>
  <nav>
    <header
      class="css-xmxh5i"
    >
      <div
        class="chakra-stack css-83r2v3"
      >
        <div>
          <div
            class="bold css-0"
            href="/"
          >
            <button
              class="chakra-button css-1a4ohpf"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-12x3nr5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <p
                class="chakra-text css-1kyf6sp"
              >
                Feed
              </p>
            </button>
          </div>
        </div>
        <div>
          <button
            class="chakra-button css-1a4ohpf"
            href="/auth/undefined"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <img
                aria-hidden="true"
                class="chakra-image__placeholder css-6fak1j"
                focusable="false"
                src="https://picsum.photos/200"
              />
            </span>
            <p
              class="chakra-text css-1lv3xn0"
            />
          </button>
        </div>
        <div>
          <div
            class="bold css-0"
          >
            <form>
              <div
                class="chakra-input__group css-4302v8"
              >
                <div
                  class="chakra-input__left-element css-1u2k0bl"
                >
                  <svg
                    class="chakra-icon css-onkibi"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <input
                  class="chakra-input css-yelcpu"
                  placeholder="Search"
                  type="text"
                />
                <div
                  class="chakra-input__right-element css-1uc557g"
                >
                  <button
                    class="chakra-button css-g7mvkz"
                    type="submit"
                  >
                    <svg
                      class="chakra-icon css-onkibi"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        fill="currentColor"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="chakra-stack css-83r2v3"
      >
        <div>
          <button
            class="chakra-button css-1a4ohpf"
            href="/create"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-j4318f"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="2"
                >
                  <rect
                    height="18"
                    rx="2"
                    ry="2"
                    width="18"
                    x="3"
                    y="3"
                  />
                  <path
                    d="M12 8v8"
                  />
                  <path
                    d="M8 12h8"
                  />
                </g>
              </svg>
            </span>
            <p
              class="chakra-text css-6iqdsv"
            >
              Add
            </p>
          </button>
        </div>
        <div>
          <button
            class="chakra-button css-1a4ohpf"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-nygapn"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  fill="currentColor"
                >
                  <path
                    d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm0,22A10,10,0,1,1,22,12,10.011,10.011,0,0,1,12,22Z"
                  />
                  <path
                    d="M17.134,15.81,12.5,11.561V6.5a1,1,0,0,0-2,0V12a1,1,0,0,0,.324.738l4.959,4.545a1.01,1.01,0,0,0,1.413-.061A1,1,0,0,0,17.134,15.81Z"
                  />
                </g>
              </svg>
            </span>
            <p
              class="chakra-text css-1749hm6"
              href="/drafts"
            />
          </button>
        </div>
        <div>
          <button
            class="chakra-button css-1a4ohpf"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-1p86u1z"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.5,9.5h-.75V6.75a6.75,6.75,0,0,0-13.5,0V9.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5ZM7.75,6.75a4.25,4.25,0,0,1,8.5,0V9a.5.5,0,0,1-.5.5H8.25a.5.5,0,0,1-.5-.5Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <p
              class="chakra-text css-1ip116p"
            >
              Log out
            </p>
          </button>
        </div>
        <div>
          <div
            class="css-17hs7hi"
          >
            <button
              class="chakra-button css-5c2176"
              type="button"
            >
              <svg
                class="chakra-icon css-nyh0wu"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="5"
                  />
                  <path
                    d="M12 1v2"
                  />
                  <path
                    d="M12 21v2"
                  />
                  <path
                    d="M4.22 4.22l1.42 1.42"
                  />
                  <path
                    d="M18.36 18.36l1.42 1.42"
                  />
                  <path
                    d="M1 12h2"
                  />
                  <path
                    d="M21 12h2"
                  />
                  <path
                    d="M4.22 19.78l1.42-1.42"
                  />
                  <path
                    d="M18.36 5.64l1.42-1.42"
                  />
                </g>
              </svg>
            </button>
          </div>
        </div>
      </div>
    </header>
  </nav>
  <div
    class="css-11rrnlf"
  />
</div>
`;
